provider/azure/metadata_linux.go

```diff
@@ -0,0 +1,46 @@
+// +build linux
+
+package azure
+
+import (
+	"encoding/base64"
+	"encoding/xml"
+	"io/ioutil"
+)
+
+type OvfEnv struct {
+	XMLName             xml.Name            `xml:"Environment"`
+	ProvisioningSection ProvisioningSection `xml:"ProvisioningSection"`
+}
+
+type ProvisioningSection struct {
+	XMLName                           xml.Name                          `xml:"ProvisioningSection"`
+	LinuxProvisioningConfigurationSet LinuxProvisioningConfigurationSet `xml:"LinuxProvisioningConfigurationSet"`
+}
+
+type LinuxProvisioningConfigurationSet struct {
+	XMLName    xml.Name `xml:"LinuxProvisioningConfigurationSet"`
+	CustomData string   `xml:"CustomData"`
+}
+
+var customDataPath = "/var/lib/waagent/ovf-env.xml"
```

Comment:
Do we need to do anything special in the deployment of start-worker to ensure it runs after the windows azure agent?

Reply:
I think we can add a service dependency in NSSM like so `nssm set worker-runner DependOnService WindowsAzureGuestAgent`. I found the service name [here](https://docs.microsoft.com/en-us/archive/blogs/mast/vm-agent-available-for-azure-vms).

Reply:
Sounds good.  There's a deployment doc in the docs directory that would be a good place to put that information.